src/ReplicatedStorage/Client/InstanceComponents/Misc/Sprint.lua

```diff
@@ -0,0 +1,63 @@
+local Players = game:GetService("Players")
+local ReplicatedStorage = game:GetService("ReplicatedStorage")
+local ContextActionService = game:GetService("ContextActionService")
+local Trove = require(ReplicatedStorage.Packages.Trove)
+local Comm = require(ReplicatedStorage.Packages.Comm)
+
+local Sprint = {}
+Sprint.__index = Sprint
+
+function Sprint.new(character: Model)
+	local clientComm = Comm.ClientComm.new(character, false, "Sprint")
+	local self = setmetatable({}, Sprint)
+
+	self._character = character
+	self._comm = clientComm
+	self._trove = Trove.new()
+	self._trove:Add(self._comm)
+	self._serverObject = clientComm:BuildObject()
+
+	self._owner = Players:GetPlayerFromCharacter(character.Parent)
+	character.AncestryChanged:Connect(function()
+		self._owner = Players:GetPlayerFromCharacter(character.Parent)
+	end)
+	if not self:IsOwner() then
+		return self
+	end
+
+	ContextActionService:BindAction("Sprint",
+		function(actionName: string, userInputState: Enum.UserInputState)
+			local humanoid = character:FindFirstChildWhichIsA("Humanoid")
+
+			if userInputState == Enum.UserInputState.Begin and humanoid then
+				self:StartSprinting()
+			elseif userInputState == Enum.UserInputState.End then
+				self:StopSprinting()
+			end
+
+			return Enum.ContextActionResult.Pass
+		end,
+	false, Enum.KeyCode.LeftShift)
```

Comment:
Just making a note, but, eventually, we should probably consider gamepad/controller support and possibly mobile support here. We will probably want more refined action bindings for this to be a thing though.

P.s. my jetpack code has this too, but, I could really be using the dedicated enum for the jump action.